owncloud/core
# [stable10] Backport of new tag editable added to systemtag

Added new tag editable to the systemtag.
This tag could be editable only to admins
and to the groups who have permission to edit.
For all other users, this tag could be used
to assign/unassign.

Signed-off-by: Sujith H <sharidasan@owncloud.com>



## Description

Introducing new tags known as `Static` tags. This tag can only renamed or deleted by admin user. This tag can only be assigned/unassigned by users of a group which is whitelisted in the static tags settings. Any other users can only see the tag and cannot assign/unassign.

## Related Issue




- Fixes <issue_link>

## Motivation and Context


## How Has This Been Tested?



- Create admin user followed by users `user1`, `user2`, `user3` and `user4`.
- Create 2 groups `group1` and `group2`. `user1` and `user2` belong to `group1`. And `user3` belong to `group2`. `user4` belongs to `group1` and `group2`.
- Now create 4 tags 
    - `visibleTag` of type `Visible`
    - `restrictedTag` of type Restricted and add `group1` to it
    - `staticTag` of type `Static` and add `group2` to it
    - `invisibleTag` of type Invisible
- Login to `user1` and `user2` , the users will see `restrictedTag` and `visibleTag`.
- Login to `user3`, the users will see `staticTag` and `visibleTag`
- Login to `user4`, the users will see `staticTag`, `restrictedTag` and `visibleTag`
- Create a file in `user4` and share it to `group1`. Add `restrictTag` and `staticTag` to the file.
- Login back to `user1` and verify that `staticTag` is seen in the files view. But it cannot be edited or assigned/unassigned by `user1`.
- Share the file created in `user4` to `group2`. The same file is now shared to `group1` and `group2`.
- Now login back to `user3`. The `restrictedTag` is seen in the files view. The `staticTag` is now assignable/unassignable.
- Login back to the `admin` user, all the tags are editable.

## Screenshots (if appropriate):

## Types of changes


## Checklist:



## Open tasks:
